timed/employment/views.py

```diff
@@ -92,20 +96,21 @@ def get_queryset(self):
                     Q(task__in=assigned_tasks) | Q(user=user)
                 )
                 return queryset.filter(Q(reports__in=visible_reports) | Q(id=user.id))
-            raise exceptions.PermissionDenied("User has no employment")
+            msg = "User has no employment"
+            raise exceptions.PermissionDenied(msg) from None
+        else:
+            return queryset
```

Comment:
This is weird control flow. This should be in the `try` block, where it belongs to the `if` on line 70, as it was before.

The way it is now makes the code unneccessarily hard to follow

Reply:
https://docs.astral.sh/ruff/rules/try-consider-else/

Reply:
Yeah one can discuss it in the simple case as in the docs, but here you're still returning *sometimes* from the `try` block, which this rule doesn't catch. You're basically moving a non-exception-related bit of code to the exception structure, which makes things confusing.

Looking at the code again - what you *should* do is to move all the code out of the try/except that doesn't actually need to be there.

```python
        try:
            current_employment = models.Employment.objects.get_at(
                user=user, date=datetime.date.today()
            )
        except models.Employment.DoesNotExist:
            if CustomerAssignee.objects.filter(user=user, is_customer=True).exists():
                assigned_tasks = Task.objects.filter(
                    Q(
                        project__customer__customer_assignees__user=user,
                        project__customer__customer_assignees__is_customer=True,
                    )
                )
                visible_reports = Report.objects.all().filter(
                    Q(task__in=assigned_tasks) | Q(user=user)
                )
                return queryset.filter(Q(reports__in=visible_reports) | Q(id=user.id))
            msg = "User has no employment"
            raise exceptions.PermissionDenied(msg) from None

        if current_employment.is_external:
            assigned_tasks = Task.objects.filter(
                Q(task_assignees__user=user, task_assignees__is_reviewer=True)
                | Q(
                    project__project_assignees__user=user,
                    project__project_assignees__is_reviewer=True,
                )
                | Q(
                    project__customer__customer_assignees__user=user,
                    project__customer__customer_assignees__is_reviewer=True,
                )
            )
            visible_reports = Report.objects.all().filter(
                Q(task__in=assigned_tasks) | Q(user=user)
            )
            return queryset.filter(Q(reports__in=visible_reports) | Q(id=user.id))
    else:
        return queryset

```

Reply:
this is so nice, while the ruff check didn't point us towards the exact direction of a proper fix. the rule helped surface a code smell that we were able to remediate, leading to a more correct implementation in this method

i think that is quite neat :mechanical_arm: